### DISCUSSION TITLE
Does H.NotifyIcon adapt to display scale change?

### DISCUSSION THREAD

**Discussion by myfix16** [HavenDV/H.NotifyIcon#76] [Q&A]
I use the following code to set up the tray icon：
```
<tb:TaskbarIcon.GeneratedIcon>
        <tb:GeneratedIcon
            Text="99"
            Foreground="White"
            FontSize="40"
            FontFamily="Arial"
            FontWeight="SemiBold"
        />
</tb:TaskbarIcon.GeneratedIcon>
```
When I set display scale to 200%, the icon fits the task bar well:
<img width="117" alt="image" src="https://user-images.githubusercontent.com/61649477/219876684-60595901-b8f9-4d78-aed7-c140edd93b76.png">

However, when I set display scale to 150%, the icon looks too small:
<img width="120" alt="image" src="https://user-images.githubusercontent.com/61649477/219876661-9703bd4f-8d7d-4883-a9ef-228181a11f36.png">

Does H.NotifyIcon adapts to display scale change?

**Comment by HavenDV** [comment]
This is rather strange, given that the icon has a fixed size when drawn (128x128).
Try setting the Size property to 256 or 512 (increasing FontSize as needed)
Related code:
https://github.com/HavenDV/H.NotifyIcon/blob/master/src/libs/H.NotifyIcon/IconGenerator.cs
https://github.com/HavenDV/H.NotifyIcon/blob/master/src/libs/H.NotifyIcon.Shared/GeneratedIcon.cs

**Reply by myfix16** [reply]
@HavenDV  I think the problem is similar to that described in https://github.com/HavenDV/H.NotifyIcon/issues/32#issue-1329333318. I use the proper app manifest, but the problem still occurs.

**Comment by myfix16** [comment]
I did an experiment and found that, with default icon size, to have the same visual size, the font size needs to be set to the following numbers:

{ 1.0, 82 },
{ 1.25, 66 },
{ 1.5, 55 },
{ 1.75, 45 },
{ 2.0, 40 },
{ 2.25, 36 },
{ 2.5, 33 },
{ 3, 27 },
{ 3.5, 23 },

The first number is display scale (1.5 means 150%, etc), and the second number is max font size.
